core/clients.py

```diff
@@ -751,6 +751,11 @@ async def get_user_info(self) -> Optional[dict]:
                 }
             }
 
+    async def get_guild_icon(self):
+        if self.bot.guild.icon is None:
+            return self.bot.user.display_avatar.url
+        return self.bot.guild.icon.url
```

Comment:
To be honest, this should not be `async`.

But if this is changed, well, you have to remove all `await`s on the changes you made.